x-pack/plugins/cases/server/client/cases/metrics/index.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MetricsResponse, MetricsResponseRt } from '../../../../common';
+import { Operations } from '../../../authorization';
+import { createCaseError } from '../../../common';
+import { CasesClient } from '../../client';
+import { CasesClientArgs } from '../../types';
+import { AlertsCount } from './alerts_count';
+import { AlertDetails } from './alert_details';
+import { Connectors } from './connectors';
+import { Lifespan } from './lifespan';
+import { MetricsHandler } from './types';
+
+export interface MetricsParams {
+  /**
+   * The ID of the case.
+   */
+  caseId: string;
+  /**
+   * The metrics to retrieve.
+   */
+  features: string[];
+}
+
+export const getMetrics = async (
+  params: MetricsParams,
+  casesClient: CasesClient,
+  clientArgs: CasesClientArgs
+): Promise<MetricsResponse> => {
+  const handlers = buildHandlers(params, casesClient, clientArgs);
+  checkAndThrowIfInvalidFeatures(params, handlers, clientArgs);
+  await checkAuthorization(params, clientArgs);
```

Comment:
I think that this should be the first check.

Reply:
Yeah that's fine, I don't think we're consistent across the other APIs, I believe some with throw an error if a parameter is invalid before checking authorization.